docs/examples/hairpin_fitting/hairpin_unfolding.rst

```diff
@@ -0,0 +1,339 @@
+    )
+
+In this example, we fit a polynomial function with `degree=1`, which is a linear function. 
+Plot the result of the fit::
+
+    plt.figure()
+    plt.title("Mirror 1 to Distance calibration")
+    distance_calibration.plot()
+
+.. image:: calibration.png
+
+Force Baseline Calibration
+--------------------------
+
+Load and plot the baseline data::
+
+    baseline_data = lk.File("test_data/FDCURV~1.H5")
+
+    baseline_1x_data = baseline_data["Force LF"]["Force 1x"]
+    baseline_2x_data = baseline_data["Force LF"]["Force 2x"]
+    distance = baseline_data["Distance"]["Distance 1"]
+
+    plt.figure()
+    plt.plot(distance.data, baseline_1x_data.data, label = "Baseline 1x")
+    plt.plot(distance.data, baseline_2x_data.data, label = "Baseline 2x")
+    plt.legend()
+    plt.ylabel("Force (pN)")
+    plt.xlabel(r"Distance ($\mu$m)")
+
+.. image:: baselines.png
+
+If the force was not reset before recording the baseline, it is best to subtract it before fitting. 
+The force offset can be determined by measuring the force when the traps are far and no force is applied, which corresponds to the first seconds in the plot below::
+
+    plt.figure()
+    plt.subplot(2,1,1)
+    baseline_1x_data.plot(label = "Baseline 1x")
+    baseline_2x_data.plot(label = "Baseline 2x")
+    plt.legend()
+    plt.ylabel("Force (pN)")
+    plt.subplot(2,1,2)
+    distance.plot()
+    plt.ylabel(r"Distance ($\mu$m)")
+    plt.tight_layout()
+
+.. image:: baseline_vs_time.png
+
+Below, we average the force at large distance to estimate the distance offset::
+
+    tmin_offset = "0s"
+    tmax_offset = "1s"
+
+    baseline_1x_data_hf = baseline_data["Force HF"]["Force 1x"]
+    baseline_2x_data_hf = baseline_data["Force HF"]["Force 2x"]
+
+    f1_offset = np.mean(baseline_1x_data_hf[tmin_offset:tmax_offset].data)
+    f2_offset = np.mean(baseline_2x_data_hf[tmin_offset:tmax_offset].data)
+
+    baseline_1x_no_offset = baseline_1x_data_hf - f1_offset
+    baseline_2x_no_offset = baseline_2x_data_hf - f2_offset
+
+Fit the baselines using a 7th degree polynomial function::
+
+    baseline_1x = lk.ForceBaseLine.polynomial_baseline(
+    baseline_data["Trap position"]["1X"], baseline_1x_no_offset, degree=7, downsampling_factor=500
+    )
+    baseline_2x = lk.ForceBaseLine.polynomial_baseline(
+    baseline_data["Trap position"]["1X"], baseline_2x_no_offset, degree=7, downsampling_factor=500
+    )
+
+Fit the result of the fit::
+
+    plt.figure()
+    baseline_1x.plot(label="baseline f1x")
+    baseline_2x.plot(label="baseline f2x")
+    plt.ylabel("Force (pN)")
+    plt.legend()
+
+.. image:: baselinefit.png
+
+The quality of the fit can be visualized by plotting the residuals. 
+When the `degree`` of the fitted polynomial is too low, the residuals will be large and not flat. When the degree is too high, the baseline fit above will give a warning that the fit may be poorly conditioned::
```

Comment:
```suggestion
When the `degree`` of the fitted polynomial is too low, the residuals will be large and not flat.::
```
I don't think the warning that it is poorly conditioned necessarily means the degree is too high. It says something about the estimability of the parameters. The data description might still be fine.

In this case we're not extrapolating, nor are we attaching any sort of interpretation to these interpolant parameters themselves. If you want to check whether the fit is negatively affected by ill-conditioning, one way to do this would be to cross validate. Remove part of the data (maybe every other point?) and see if the error on those increases when the degree increases.

Unfortunately, I don't think the way we set up the API allows users to do this easily, so for now, I would only remove that second statement. I'll make a ticket on our to-do list to offer some sort of cross-validation-based check.